albogdan/skule-vote
# Connected to UofT Auth Service

## Overview

- Resolves #14 
- Removed logout button in the frontend
- Linked the vote button in the frontend to uoft endpoint
- IN PROGRESS: setup the redirect chain for user authentication
- Fixed various typos
- Added more error checks for incorrect/invalid cookies
- Changed the column width for student number hash in db

** Additional change **
- Changes a line in the default nginx config to allow it to reverse-proxy sub-urls for the react app

## Unit Tests Created

- None


## Steps to QA

- Set env vars: `export CONNECT_TO_UOFT=1`, `export UOFT_SECRET_KEY=ask armin`
- Go to frontend and make sure the vote buttons in the navbar and on the page redirect to https://portal.engineering.utoronto.ca/weblogin/sites/apsc/vote.asp
- After logging in at the above URL, make sure you get redirected to https://vote.skule.ca/vote followed by a long query string. This should be a 404 error
- Without changing the query string, change `https://vote.skule.ca/vote` to `http://localhost:8000/vote` and go to that url
- You should end up in `/api/elections` with a list of on going elections that you are eligible to vote in as the response 
- In the admin panel you should see a voter objected created for you with the correct information about your year, discipline, and status
- Go to `localhost:3000/elections`. You should see any ongoing eligible elections. 
- Click on "check eligibility". It should say you are an eligible student
